pkgs/development/libraries/grpc/default.nix

```diff
@@ -27,8 +29,9 @@ stdenv.mkDerivation rec {
       "-DgRPC_SSL_PROVIDER=package"
       "-DgRPC_PROTOBUF_PROVIDER=package"
       "-DgRPC_GFLAGS_PROVIDER=package"
-      "-DBUILD_SHARED_LIBS=ON"
+      "-DBUILD_SHARED_LIBS=${if enableShared then "ON" else "OFF"}"
```

Comment:
This should not be necessary since the `makeStaticLibraries` adapter appends `-DBUILD_SHARED_LIBS=OFF` already.